internal/apis/meta/v1/conversion.go

```diff
@@ -20,7 +20,7 @@ import (
 	"k8s.io/apimachinery/pkg/conversion"
 
 	"github.com/jetstack/cert-manager/internal/apis/meta"
-	"github.com/jetstack/cert-manager/pkg/apis/meta/v1"
+	v1 "github.com/jetstack/cert-manager/pkg/apis/meta/v1"
```

Comment:
I don't think we want this change, or the others like it? It's a redundant alias (and my IDE highlights it as such..)

Maybe there's some extra options that can be passed to `goimports` to stop it doing this? How come these packages specifically do it?

Reply:
+1, this looks weird / makes no difference at all, so is this a bug in goimports?

Reply:
We [document](https://cert-manager.io/docs/contributing/coding-conventions/) that we use `goimports` but not which flags. There aren't really any relevant flags, though:

```console
$ goimports --help
usage: goimports [flags] [path ...]
  -cpuprofile string
        CPU profile output
  -d    display diffs instead of rewriting files
  -e    report all errors (not just the first 10 on different lines)
  -format-only
        if true, don't fix imports and only format. In this mode, goimports is effectively gofmt, with the addition that imports are grouped into sections.
  -l    list files whose formatting differs from goimport's
  -local string
        put imports beginning with this string after 3rd-party packages; comma-separated list
  -memprofile string
        memory profile output
  -memrate int
        if > 0, sets runtime.MemProfileRate
  -srcdir dir
        choose imports as if source code is from dir. When operating on a single file, dir may instead be the complete file name.
  -trace string
        trace profile output
  -v    verbose logging
  -w    write result to (source) file instead of stdout
```

It looks like this is actually a performance improvement that `goimports` [sorta does for itself](https://go-review.googlesource.com/c/tools/+/152000/). See also [this go issue](https://github.com/golang/go/issues/28428).

As far as I'm concerned, if `goimports` says it's wrong then it's wrong. It's moot anyway; I think Irbe's suggestion to use `cmmeta` instead for consistency is clearly the way to go.

Reply:
Not sure if that's the correct issue for it - that mentions the case where the directory name does NOT match the name of the package. In this instance, the package name does match the name of the directory so I don't think it's the same thing.. unless goimports for some reason thinks the package is named differently 🤔

Reply:
https://github.com/golang/go/issues/29041#issuecomment-443275406

It's an edge case in the tooling and the accepted solution is goimports will explicitly add named imports `v(num)` packages. I think this is dumb, but the advantage of go is the opinionated tooling is designed to end all arguments about the "correct" way to do things!

Reply:
I shouldn't have muddied the waters linking to that second issue which contains a revert. [The first one I linked](https://go-review.googlesource.com/c/tools/+/152000/) looks to me to be the correct issue